src/software/thunderscope/robot_communication.py

```diff
@@ -10,32 +12,51 @@ class RobotCommunication(object):
 
     def __init__(
         self,
-        proto_unix_io,
+        full_system_proto_unix_io,
         multicast_channel,
         interface,
         estop_path="/dev/ttyACM0",
         estop_buadrate=115200,
     ):
         """Initialize the communication with the robots
 
-        :param proto_unix_io: proto_unix_io object
+        :param full_system_proto_unix_io: full_system_proto_unix_io object
         :param multicast_channel: The multicast channel to use
         :param interface: The interface to use
         :param estop_path: The path to the estop
         :param estop_baudrate: The baudrate of the estop
 
         """
-        self.proto_unix_io = proto_unix_io
-        self.multicast_channel = multicast_channel
+        self.sequence_number = 0
+        self.last_time = time.time()
+        self.full_system_proto_unix_io = full_system_proto_unix_io
+        self.multicast_channel = str(multicast_channel)
         self.interface = interface
         self.estop_path = estop_path
         self.estop_buadrate = estop_buadrate
 
+        self.robots_connected_to_handheld_controllers = set()
+        self.robots_connected_to_diagnostics = set()
+
         self.world_buffer = ThreadSafeBuffer(1, World)
         self.primitive_buffer = ThreadSafeBuffer(1, PrimitiveSet)
 
-        self.proto_unix_io.register_observer(World, self.world_buffer)
-        self.proto_unix_io.register_observer(PrimitiveSet, self.primitive_buffer)
+        self.motor_control_diagnostics_buffer = ThreadSafeBuffer(1, MotorControl)
+        self.power_control_diagnostics_buffer = ThreadSafeBuffer(1, PowerControl)
```

Comment:
Might need a `HRVOVisualization` buffer as well. Same with a `HRVOVisualizationProtoListener` lower down.

Reply:
doesn't need to be included in this PR, I'll add it later on as part of the robot visualization from robot PR